src/DataStructures/LeviCivitaIterator.cpp

```diff
@@ -0,0 +1,65 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "DataStructures/LeviCivitaIterator.hpp"
+
+#include "Utilities/Gsl.hpp"
+
+namespace {
+template <size_t Dim>
+constexpr void initialize_indexes_and_signs(
+    gsl::not_null<std::array<Index<Dim>, factorial(Dim)>*> indexes,
+    gsl::not_null<std::array<int, factorial(Dim)>*> signs) noexcept {
+  auto index = Index<Dim>();
+  std::iota(index.begin(), index.end(), 0);
+
+  // Add the initial index permutation to the list
+  // Also add the sign, which is always +1, because std::iota
+  // initializes the indexes as 0,1,...Dim-1
+  (*indexes)[0] = index;
+  (*signs)[0] = 1;
+
+  size_t permutation_count = 1;
+  auto factoradic_counter = std::array<size_t, Dim>();
+
+  // std::next_permutation generates the different permuations of index,
+  // but how do you know whether the corresponding Levi Civita symbol is
+  // +1 or -1? To find out, compute the number, in the factorial number system,
+  // corresponding to each permutation, and sum the digits. If the sum is
+  // even (odd), the corresponding Levi-Civita symbol will be +1 (-1).
+  // This works because there are Dim! unique permutations of the
+  // Levi Civita symbol indices, so each one can be represented by a
+  // (`Dim`)-digit factorial-number-system number. For more on the
+  // factorial number system, see
+  // https://en.wikipedia.org/wiki/Factorial_number_system
+  for (; std::next_permutation(index.begin(), index.end());) {
+    for (size_t i = 0; i < Dim; ++i) {
+      factoradic_counter[i]++;
+      if (factoradic_counter[i] < i + 1) {
+        break;
+      } else {
+        factoradic_counter[i] = 0;
+      }
+    }
+
+    (*indexes)[permutation_count] = index;
+    (*signs)[permutation_count] =
+        ((std::accumulate(factoradic_counter.begin(), factoradic_counter.end(),
+                          0) %
```

Comment:
Hopefully this can be formatted better

Reply:
This is clang-formatted already. What would you prefer?

Reply:
I thought something could be done without having to wrap the code (i.e. write the code in such a way that it ends up clang-formatted differently) but I was wrong - I'm ok leaving the current code as is.